### PR TITLE
Add getActiveAttrib

### PR DESCRIPTION
**Pull request by dzbarsky**
This change is [Reviewable](https://reviewable.io/reviews/servo/gleam/73)

### CODE REVIEW

**Comment by dzbarsky** [comment]
@emilio r?

File: src/gl.rs [hunk 1]
```diff
@@ -970,6 +970,20 @@ pub fn depth_range(near: f64, far: f64) {
...
 
 #[inline]
+pub fn get_active_attrib(program: GLuint, index: GLuint) -> (i32, u32, String) {
+    let buf_size = get_program_iv(program, ffi::ACTIVE_ATTRIBUTE_MAX_LENGTH);
+    let mut name: Vec<_> = repeat(0u8).take(buf_size as usize).collect();
+    let mut length = 0 as GLsizei;
+    let mut size = 0 as i32;
+    let mut type_ = 0 as u32;
+    unsafe {
+        ffi::GetActiveAttrib(program, index, buf_size, &mut length, &mut size, &mut type_, name.as_ptr() as *mut GLchar);
```
Comment (jdm) [inline src/gl.rs]: `name.as_mut_ptr()` would make more sense.

File: src/gl.rs [hunk 2]
```diff
@@ -970,6 +970,20 @@ pub fn depth_range(near: f64, far: f64) {
 }
 
 #[inline]
+pub fn get_active_attrib(program: GLuint, index: GLuint) -> (i32, u32, String) {
+    let buf_size = get_program_iv(program, ffi::ACTIVE_ATTRIBUTE_MAX_LENGTH);
+    let mut name: Vec<_> = repeat(0u8).take(buf_size as usize).collect();
```
Comment (emilio) [inline src/gl.rs]: `vec![0u8; buf_size]` might be cleaner?

File: src/gl.rs [hunk 3]
```diff
@@ -970,6 +970,20 @@ pub fn depth_range(near: f64, far: f64) {
...
+    let mut length = 0 as GLsizei;
+    let mut size = 0 as i32;
+    let mut type_ = 0 as u32;
+    unsafe {
+        ffi::GetActiveAttrib(program, index, buf_size, &mut length, &mut size, &mut type_, name.as_ptr() as *mut GLchar);
+    }
+    name.truncate(if length > 0 {length as usize - 1} else {0});
```
Comment (emilio) [inline src/gl.rs]: Why `length - 1`? 

Per the OpenGL docs:

> Returns the size of the attribute variable.

Doesn't seem to count the trailing null byte or anything like that.

File: src/gl.rs (hunk 2 above)
Comment (dzbarsky) [inline src/gl.rs]: i copied get_[program|shader]_info_log, happy to change them all but let's be consistent

File: src/gl.rs (hunk 1 above)
Comment (dzbarsky) [inline src/gl.rs]: will change

File: src/gl.rs (hunk 3 above)
Comment (dzbarsky) [inline src/gl.rs]: Good point, it doesn't count the null.

Comment (emilio) [inline src/gl.rs]: `get_*_info_log` seem to do this also, even though the docs[1] explicitely state *excluding the null terminator*, so I'm either missing something or those should be changed too.

[1]: https://www.khronos.org/opengles/sdk/docs/man/xhtml/glGetProgramInfoLog.xml

File: src/gl.rs (hunk 2 above)
Comment (emilio) [inline src/gl.rs]: Yeah, I'm pro-changing that.

File: src/gl.rs (hunk 1 above)
Comment (emilio) [inline src/gl.rs]: Yeah, nontheless using `as_mut_ptr` to mutate the `Vec` makes a lot more sense, so I'd prefer `name.as_mut_ptr() as *mut _`.

**Comment by emilio** [comment]
@bors-servo: r+

Thanks! :)

**Pull request merged**